src/Microsoft.DotNet.Build.Tasks.Packaging/src/GenerateNuSpec.cs

```diff
@@ -69,6 +69,8 @@ public class GenerateNuSpec : Task
 
         public string Tags { get; set; }
 
+        public ITaskItem[] PackageTypes { get; set; }
```

Comment:
The real pack task takes a `string[]` instead: https://github.com/NuGet/NuGet.Client/blob/50af5271b98ac5cb2896a707569bc4cd1e87a017/src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs#L22

Reply:
Ok, so why aren’t you doing that if you don’t need items?

Reply:
The real pack task takes the raw item strings and does custom parsing on it. Using `ITaskItem[]` for our own code makes it simpler without needing to duplicate the `ParsePackageTypes` method in `PackTaskLogic.cs`. If we actually need the more complex logic later on, our time would probably be better invested in just moving to use the official task.

Reply:
How is our code made simpler by taking `ITaskItem[]`?  I see how it is made simpler by not supporting the encoded version in the identity string, but that doesn't have anything to do with accepting `ITaskItem[]` vs `string[]`.  Seems to me the code is made *slightly* more complex by having to get the `ItemSpec` out instead of just using the value.

Typically (though I know we don't follow this guideline everywhere) it is better/cheaper/simpler to accept `string` if you have no need for metadata, path semantics, etc provided by `ITaskItem`.

Reply:
@ericstj stopped by and we chatted. I was under the impression that using `ITaskItem[]` was the general/recommended approach for simplicitly/cost. However, that may not always be the case.

In either case, I took another look at the way NuGet expects the MSBuild items to be encoded, it appears I was mistaken.

Unlike the `nuspec`, which represents these as `<packageType name="Dependency" version="Version" />`, MSBuild is expected to encode it as `<PackageType>Dependency1,Version2;Dependency2,Version2</PackageType>` (`<PackageType Include="Dependency" Version="Version" />` is not supported). So we ultimately need to just take a `string[]` anyways.

Reply:
For reference: https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets and https://github.com/NuGet/Home/wiki/Package-Type-%5BPacking%5D